app/main.py

```diff
@@ -1 +1,49 @@
-# write your code here
+class SoftwareEngineer:
+    def __init__(self, name: str) -> None:
+        self.name = name
+        self.skills = []
+
+    def learn_skill(self, skill: str) -> None:
+        self.skills.append(skill)
+
+
+class FrontendDeveloper(SoftwareEngineer):
+    def __init__(self, name: str) -> None:
+        super().__init__(name)
+        self.skills = ["JavaScript", "HTML", "CSS"]
```

Comment:
each subclass (FrontendDeveloper, BackendDeveloper, AndroidDeveloper) assigns a new list of skills directly to self.skills:

self.skills = ["JavaScript", "HTML", "CSS"]

Problem: This overrides the skills list that might have been previously populated (e.g., via the learn_skill method in the base SoftwareEngineer class), meaning any learned skills will be lost.

Fix: The correct way is to extend the existing skills list using extend():